### PR TITLE
make windows from RectangularZoneTemplate.mo replaceable and correct …

### LINKED ISSUE #890: correct surface area subtraction in template for nWin
correct surface area subtraction in template for nWin

### PR DESCRIPTION
**Pull request by damienpicard**
This closes #890 and #891

### CODE REVIEW

**Comment by damienpicard** [comment]
I have also slightly changed the GUI of the rectangularZoneTemplate: now each propsBus for wall is the the left corner.

**Comment by damienpicard** [comment]
This is not yet ready to be merged. I get the following error:
Undeclared variable: zoneIntWal.winB.A since the declaration of zoneIntWal.winB was conditionally removed
in the definition equation
zoneIntWal.outB.A = zoneIntWal.lB*zoneIntWal.h-(if zoneIntWal.hasWinB then zoneIntWal.winB.A* zoneIntWal.winB.nWin else 0)

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 1]
```diff
@@ -1281,6 +1281,12 @@ initial equation
   AWallD = lD*h - (if hasWinD then  propsBusInt[indWinD].area else 0);
   ACei = A - (if hasWinCei then  propsBusInt[indWinCei].area else 0);
 
+  // assert that the windows are not bigger than the wall
+  assert(AWallA >= 0, "The netto surface area of wall A is negative. This is not allowed.");
+  assert(AWallB >= 0, "The netto surface area of wall B is negative. This is not allowed.");
+  assert(AWallD >= 0, "The netto surface area of wall C is negative. This is not allowed.");
+  assert(AWallD >= 0, "The netto surface area of wall D is negative. This is not allowed.");
+  assert(ACei >= 0, "The netto surface area of the ceiling is negative. This is not allowed.");
```
Comment (damienpicard) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: Surprisingly, when AWallA becomes negative, no assert is triggered. I only get in the simulation tab:
```
dymosim started
... "dsin.txt" loading (dymosim input file)
Error: The following error was detected at time: 0

Model error - sqrt (-2.3999999999999986) = sqrt (-2.4)

Error: Integrator failed to start model.
```

Moving the assert statement to the equation section did not help.

Comment (damienpicard) [reply IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: This can be solve by using a max(0,value) on AWall and in assert, using not ==0.

**Comment by Mathadon** [comment]
@damienpicard this looks good once unit tests have been updated and once we have split up the zone template in a partial.

**Comment by Mathadon** [comment]
@damienpicard I propose to merge the LIDEAS pull request first to avoid messing up its unit tests again when merging this pull request.

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 2]
```diff
@@ -1,446 +1,58 @@
...
-  parameter SI.Length PWall = (if hasOutA then lA else 0) + (if hasOutB then lB else 0) + (if hasOutC then lC else 0) + (if hasOutD then lD else 0)
-  "Total floor slab perimeter length" annotation(Dialog(tab="Advanced", group="SlabOnGround", enable=(bouTypFlo == IDEAS.Buildings.Components.Interfaces.BoundaryType.SlabOnGround)));
-
-  IDEAS.Buildings.Components.Interfaces.ZoneBus[nSurfExt] proBusExt(
-    each final numIncAndAziInBus=sim.numIncAndAziInBus,
-    each final outputAngles=sim.outputAngles) if nSurfExt>0
-    "Propsbus for connecting additional external surfaces" annotation (
-      Placement(transformation(
-        extent={{-20,20},{20,-20}},
+  Modelica.Blocks.Interfaces.RealInput ctrlA if
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: can't these also be added to the partial?

**Pull request merged**